### PR TITLE
tests: add cache start trim test

### PR DESCRIPTION
**Pull request by micrakow**
Signed-off-by: Michal Rakowski <michal.rakowski@intel.com>

### CODE REVIEW

File: test/functional/tests/trim/test_trim.py [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+
+    with TestRun.step("Starting cache"):
+        cache = casadm.start_cache(cas_part, force=True)
+        metadata_size = get_metadata_size_from_dmesg()
+
+    with TestRun.step("Stop blktrace and check if discard requests were issued"):
+        cache_reqs = blktrace.stop_monitoring()
+        cache_part_start = cas_part.begin
+
+        if len(cache_reqs) == 0:
```
Comment (mmichal10) [inline test/functional/tests/trim/test_trim.py:60]: Where did you triggered discard?

File: test/functional/tests/trim/test_trim.py [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+        cas_part = dev.partitions[0]
+        non_cas_part = dev.partitions[1]
+
+    with TestRun.step("Writing different pattern on partitions"):
+        cas_fio = write_pattern(cas_part.system_path)
+        non_cas_fio = write_pattern(non_cas_part.system_path)
+        cas_fio.run()
+        non_cas_fio.run()
+
+    # TODO add blktracing for non-cas part
```
Comment (mmichal10) [inline test/functional/tests/trim/test_trim.py:47]: Wouldn't it be just copy-paste of blktracing cas part?

File: test/functional/tests/conftest.py [hunk 3]
```diff
@@ -169,8 +169,7 @@ def base_prepare(item):
 
         for disk in TestRun.dut.disks:
             disk.umount_all_partitions()
-            if not create_partition_table(disk, PartitionTable.gpt):
-                raise Exception(f"Failed to remove partitions from {disk}")
+            create_partition_table(disk)
```
Comment (mmichal10) [inline test/functional/tests/conftest.py:172]: Why did you change it? You could add it as separate commit with explanation in commit message

File: test/functional/tests/trim/test_trim.py [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+from core.test_run import TestRun
+from test_utils import os_utils
+from test_utils.size import Size, Unit
+from test_tools.fio.fio import Fio
+from test_tools.blktrace import BlkTrace, BlkTraceMask
+from test_tools.fio.fio_param import ReadWrite, IoEngine
+from storage_devices.disk import DiskType, DiskTypeSet, DiskTypeLowerThan
+
+
+#TODO Add mark for getting discard-supproting devcie
```
Comment (mmichal10) [inline test/functional/tests/trim/test_trim.py:19]: How about additional check to test if device supports discards? It would prevent test from failing until such mark would be added.

File: test/functional/tests/trim/test_trim.py [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+
+    with TestRun.step("Stop blktrace and check if discard requests were issued"):
+        cache_reqs = blktrace.stop_monitoring()
+        cache_part_start = cas_part.begin
+
+        if len(cache_reqs) == 0:
+            TestRun.fail("No discard requests issued to the cas partition!")
+
+        non_meta_sector = (cache_part_start + metadata_size).get_value(Unit.Blocks512)
+        non_meta_size = cas_part.size - metadata_size
```
Comment (mmichal10) [inline test/functional/tests/trim/test_trim.py:64]: You are using `.get_value(Unit.Byte)` later anyway, why don't You call it here? It would be more consistent

File: test/functional/tests/trim/test_trim.py [hunk 6]
```diff
@@ -0,0 +1,105 @@
...
+            if req.sector_number != non_meta_sector:
+                TestRun.fail(f"Discard request issued to wrong sector: {req.sector_number}, "
+                             f"expected: {non_meta_sector}")
+            if req.byte_count != non_meta_size.get_value(Unit.Byte):
+                TestRun.fail(f"Discard request issued with wrong bytes count: {req.byte_count}, "
+                             f"expected: {non_meta_size.get_value(Unit.Byte)}")
+
+        cas_fio.read_write(ReadWrite.read)
+        non_cas_fio.read_write(ReadWrite.read)
+        cas_fio.verification_with_pattern("0x00")
```
Comment (mmichal10) [inline test/functional/tests/trim/test_trim.py:75]: How about adding verification for second partition?

File: test/functional/tests/trim/test_trim.py (hunk 1 above)
Comment (micrakow) [reply test/functional/tests/trim/test_trim.py:60]: Discard was triggered by CAS itself during cache start

Comment (mmichal10) [reply test/functional/tests/trim/test_trim.py:60]: Could You please add a comment with this information?

File: test/functional/tests/conftest.py (hunk 3 above)
Comment (micrakow) [reply test/functional/tests/conftest.py:172]: I'll make it part of update-test-framework commit

File: test/functional/tests/trim/test_trim.py (hunk 6 above)
Comment (micrakow) [reply test/functional/tests/trim/test_trim.py:75]: It's being done at line :82 (after setting io dir to read at line :74)

File: test/functional/tests/trim/test_trim.py (hunk 4 above)
Comment (micrakow) [reply test/functional/tests/trim/test_trim.py:19]: Actually this todo can be removed since here cache will be optane or nand disk and it always supports trim (please correct me if I'm wrong)

File: test/functional/tests/trim/test_trim.py (hunk 2 above)
Comment (micrakow) [reply test/functional/tests/trim/test_trim.py:47]: This would require changes in BlkTrace module to allow it to monitor more than 1 device at once which is currently not possible :(. I would leave it as it is for now and I'll try to add such funcionality (if I make it till end of june ;))

File: test/functional/tests/trim/test_trim.py (hunk 6 above)
Comment (mmichal10) [reply test/functional/tests/trim/test_trim.py:75]: It is kind of tricky ¯\_(ツ)_/¯
